cranelift/filetests/src/zkasm_codegen.rs

```diff
@@ -33,13 +33,23 @@ pub fn generate_zkasm(settings: &ZkasmSettings, wasm_module: &[u8]) -> String {
 
     let mut program: Vec<String> = Vec::new();
 
-    let start_func = zkasm_environ
+    let main_func = zkasm_environ
         .info
-        .start_func
-        .expect("Must have a start function");
+        .functions
+        .iter()
+        .find(|&(_, exportable_func)| exportable_func.export_names.contains(&"main".to_string()));
+
+    let start_index = match main_func {
+        Some(index) => index.0,
+        None => zkasm_environ
+            .info
+            .start_func
+            .expect("Must have a start function"),
+    };
+
```

Comment:
I think it is not sound to replace start functions by an expected main function in this way. Start functions are part of the [WASM spec](https://webassembly.github.io/spec/core/syntax/modules.html#syntax-start) and the old code should be able to handle a module which doesn't have a `main` function but defines a start function with `(start $some_func)`. The new code would introduce the requirement that a module must have a `main` function for successful zkASM generation.

Reply:
Sounds good to me - I was a bit surprised to see this change introduced as side-effect of this PR. It would be good to document that a main function is required as entry point.

Reply:
Thanks, very good points!
I moved this into a separate PR: https://github.com/near/wasmtime/pull/244